### PR TITLE
Weaver 2

### LINKED ISSUE #126: [Feature] Handle WPS-1.x/2.x execute request as WPS-3 'job'
Route `/ows/wps` (default path) offers the WPS-1.x/2.x XML interface (maybe also its 2.0 JSON interface with #125). 

Since this route is served as a standard PyWPS service, it is possible to call an `Execute` request (either with GET/POST variants), but this will run the process directly on an API 'manager' thread instead of some dedicated celery worker thread/docker image. This both consumes the few resources for handling REST-API requests and possibly limit how fast/many job request the actual API can process, as each 'manager' thread becomes synchronously blocked/used until the corresponding process completes (could be a very long workflow). 

Even worst, because the WPS process is executed directly from the PyWPS service, no actual `job` gets registered in `Weaver` and it is therefore impossible to obtain any feedback from the WPS 3.x routes. The only access to process status is via the base PyWPS status URL, which will consume yet another API-manager thread. Finally, when the process completes, no log/exceptions/results will be available, except the bare WPS Status route.

Therefore, WPS-1/2 Execute requests should be redirected to the corresponding `/processes/{id}/jobs` call, so that the job gets created and processed using the normal celery worker that will pick it up eventually. Feature #21 must be implemented at the same time as the redirect will have to work the other way around of current implementation (ie: WPS-1/2 Execute => WPS-REST 'celery' execute). 
Without simultaneous implementation of the features, we will generate a circular loop of execute calls between implementations.

Given that redirections between WPS-1/2 and WPS-3 calls would be implemented here, features in #125 could be done at the same time as it implies similar requirements.

Relates to double-layer monitoring #21.

### PR DESCRIPTION
**Pull request by fmigneault**
## changes summary

- Weaver 2.0 with multiple items revamped into more suitable modules, such as conversion between `ogc-api`/`pywps`/`owslib`/`cwl` processes / app-packages.

- WPS 1/2 endpoint (default `/ows/wps`) allows both GET KVP and POST XML  WPS-Execute request, which are both dispatched to the same job worker operation as WPS-REST endpoint (fixes #126, relates to #125)
  - Returned response is formatted as JSON/XML according to provided `Accept` header. 
  - Added tests for each combination running a 'demo' docker application.
  - Started documenting the WPS-1/2 endpoint in the rendered OpenAPI (`/api` and `/json`) (relates to #146)

- Adjusted some tests to allow running *offline* using `WebTest.TestApp` instead of relying on live instance.
- Add Github Actions test suite execution 
- Add sample `docker-compose` setup for API / Worker config to run Docker application packages. 

- use branch `pywps-4.4` with latest developments

- many linting fixes that were not applied
- some devops patches in makefile that caused disjoint install (not pip dev install) which resulted in desync configs between 'run' and 'debug' mode of pyramid app.

Built on top of the other open PRs. 
closes #207 
closes #208
closes #209

### CODE REVIEW

**Review by dbyrns — COMMENTED** [review]
Good cleanup! I only got 4 comments.

File: docker/docker-compose.yml.example [hunk 1]
```diff
@@ -21,39 +17,48 @@ services:
...
-      - ./components/weaver/config/weaver/data_sources.json:/opt/local/src/weaver/config/data_sources.json
+      - ./components/weaver/config/weaver/data_sources.yml:/opt/local/src/weaver/config/data_sources.yml
       - ./components/weaver/config/weaver/request_options.yml:/opt/local/src/weaver/config/request_options.yml
       - ./components/weaver/config/weaver/wps_processes.yml:/opt/local/src/weaver/config/wps_processes.yml
-      - /usr/bin/docker:/usr/bin/docker   # FIXME: see above docker-sibling note
-      - ~/.docker/config.json:/root/.docker/config.json   # FIXME: see above docker-sibling note
       # WARNING: see detail in 'worker' definition
-      - ${WEAVER_WPS_WORKDIR}:${WEAVER_WPS_WORKDIR}     # FIXME: with fix WPS-execute inside worker, shouldn't be required anymore
-      - ${WEAVER_WPS_OUTPUTS_DIR}:${WEAVER_WPS_OUTPUTS_DIR}
+      - /tmp/weaver/wps-outputs:/tmp/weaver/wps-outputs
```
[the remark this answers is not in the document]
Comment (fmigneault) [reply docker/docker-compose.yml.example:32]: I think multiple containers can simultaneously mount the volumes and all write in them.

File: requirements.txt [hunk 2]
```diff
@@ -1,5 +1,4 @@
 alembic
-backports.tempfile; python_version < "3"
```
[the remark this answers is not in the document]
Comment (fmigneault) [reply requirements.txt:2]: Yes. Technically it was never properly maintained. Even the setup.py had `python_requires >=3`
Now I'm formally dropping it as most reference package do not support it anymore either. 

Changes has `Remove every Python 2 backward compatibility references and operations.` 
But I'll make it even more explicit.

File: requirements.txt [hunk 3]
```diff
@@ -47,9 +44,7 @@ requests_file
 # (https://bitbucket.org/ruamel/yaml/issues/261/error-while-installing-ruamelyaml-setuppy)
 # let cwltool fully define the version if python 2 is used
 ruamel.yaml>=0.15.78,<=0.16.5; python_version >= "3"
-ruamel.yaml<=0.16.5; python_version < "3"
 shapely
-six
```
[the remark this answers is not in the document]
Comment (fmigneault) [reply requirements.txt:46]: Yes, pywps drops support officially with geopython/pywps#574 (pywps-4.4) 
And I'm expecting for it to become harder overtime to support as every package is dropping it gradually.

File: docker/docker-compose.yml.example (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbyrns) [reply docker/docker-compose.yml.example:32]: It wasn't obvious, but I meant does every worker can write in this tmp directory safely without sub-directory name collision?

File: requirements.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (dbyrns) [reply requirements.txt:2]: Sorry I missed that part.

File: docker/docker-compose.yml.example (hunk 1 above)
[the remark this answers is not in the document]
Comment (fmigneault) [reply docker/docker-compose.yml.example:32]: Ah. For this there is no problem. Each worker is writing in `<weaver.workdir>/<jobId>/[outputs.ext]`

**Review by dbyrns — APPROVED** [review]
Thank you for your answers. Ready to merge.

**Pull request merged**